src/components/mortgage-investment-compare/index.js

```diff
@@ -23,16 +23,28 @@ import {
 const COMPOUND_FREQUENCY = 12
 
 class MortgageInvestmentCompare extends React.Component {
-  constructor () {
-    super()
+  constructor (props) {
+    super(props)
     this.calculateFV = this.calculateFV.bind(this)
     this.calculateInterestAmt = this.calculateInterestAmt.bind(this)
     this.calculateOpportunityCost = this.calculateOpportunityCost.bind(this)
     this.calculatePMT = this.calculatePMT.bind(this)
-    this.getResult = this.getResult.bind(this)
+    this.updtateResult = this.updtateResult.bind(this)
```

Comment:
```suggestion
    this.updateResult = this.updateResult.bind(this)
```

Reply:
One issue w/ this is that `updateResult` now becomes stateful. With `getResult`, we never had to worry about side effects w/ the app since it's pure. When we needed to modify state, we did so explicitly in a single place.

Reply:
please see long comment below